google-cloud-clients/google-cloud-core/src/main/java/com/google/cloud/Date.java

```diff
@@ -68,7 +70,39 @@ public static Date parseDate(String date) {
     int dayOfMonth = Integer.parseInt(matcher.group(3));
     return new Date(year, month, dayOfMonth);
   }
+  
+  /**
+  * @param date
+  * @return LocalDate
```

Comment:
Is this already used somewhere? If not it exposes a new third party dependency on the API surface, which we really want to avoid: 

https://github.com/GoogleCloudPlatform/cloud-opensource-java/blob/master/library-best-practices/JLBP-2.md

Reply:
Let's remove the LocalDate helpers.

Reply:
threetenbp is already exposed on the surface of google-cloud-core, e.g. `Duration`: https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-core/src/main/java/com/google/cloud/RetryOption.java#L23

threetenbp is specifically used throughout google-cloud-java in order to have a Java 8-like surface for date stuff while still staying on Java 7.